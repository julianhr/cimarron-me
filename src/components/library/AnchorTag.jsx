import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'


function AnchorTag({ children, url, rootStyle, isLinkRouted }) {
  if (isLinkRouted) {
    return (
      <Link
        to={url}
        css={rootStyle}
      >
        {children}
      </Link>
    )
  } else {
    return (
      <a
        href={url}
        css={rootStyle}
        rel='noopener noreferrer'
      >
        {children}
      </a>
    )
  }
}

AnchorTag.propTypes = {
  url: PropTypes.string.isRequired,
  rootStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
  isLinkRouted: PropTypes.bool,
}

AnchorTag.defaultTypes = {
  isLinkRouted: true,
  rootStyle: {},
}

export default AnchorTag
