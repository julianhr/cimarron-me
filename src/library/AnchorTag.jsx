import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'


function AnchorTag({ children, url, rootStyle, isLinkRouted, isTargetBlank }) {
  if (isLinkRouted) {
    return (
      <Link
        to={url}
        css={rootStyle}
      >
        {children}
      </Link>
    )
  } else {
    return (
      <a
        href={url}
        css={rootStyle}
        rel='noopener noreferrer'
        target={isTargetBlank ? '_blank' : null}
      >
        {children}
      </a>
    )
  }
}

AnchorTag.propTypes = {
  url: PropTypes.string.isRequired,
  rootStyle: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
  isLinkRouted: PropTypes.bool,
  isTargetBlank: PropTypes.bool,
}

AnchorTag.defaultProps = {
  isLinkRouted: true,
  rootStyle: {},
  isTargetBlank: false,
}

export default AnchorTag
