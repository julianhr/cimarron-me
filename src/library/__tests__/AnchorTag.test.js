import React from 'react'
import { render, mount, shallow } from 'enzyme'
import { create } from 'react-test-renderer'

import AnchorTag from '../AnchorTag'


const testProps = {
  url: 'http://test.com',
  rootStyle: { background: '#CBA' },
  isLinkRouted: false,
}

const renderComponent = props => create(<AnchorTag {...props} />)

describe('AnchorTag', () => {
  describe('props', () => {
    it('requires url', () => {
      const props = {...testProps}
      delete props.url
      expect(() => renderComponent(props)).toThrow()
    })

    it('rootStyle not required', () => {
      const props = {...testProps}
      delete props.rootStyle
      expect(() => renderComponent(props)).not.toThrow()

    })
    it('isLinkRouted not required', () => {
      const props = {...testProps}
      delete props.isLinkRouted
      expect(() => renderComponent(props)).not.toThrow()
    })
  })

  describe('snapshots', () => {
    it('matches with all props', () => {
      const wrapper = mount(<AnchorTag {...testProps}/>)
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })

    it('matches with required props', () => {
      const props = { url: testProps.url }
      const wrapper = mount(<AnchorTag {...props}/>)
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })
  })
})

