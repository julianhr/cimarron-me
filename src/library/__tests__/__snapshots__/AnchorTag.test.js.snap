// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnchorTag snapshots matches with all props 1`] = `
Array [
  <ForwardRef(render)
    styles={[Function]}
  />,
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AnchorTag"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="a"
    css={
      Object {
        "background": "#CBA",
      }
    }
    href="http://test.com"
    rel="noopener noreferrer"
    target={null}
  >
    <a
      className="css-10aldfz-AnchorTag"
      href="http://test.com"
      rel="noopener noreferrer"
      target={null}
    />
  </EmotionCssPropInternal>,
]
`;

exports[`AnchorTag snapshots matches with required props 1`] = `
Array [
  <ForwardRef(render)
    styles={[Function]}
  />,
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="AnchorTag"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
    css={Object {}}
    to="http://test.com"
  >
    <a
      className="css-q7jxp9-AnchorTag"
      href="/http://test.com"
      onClick={[Function]}
    />
  </EmotionCssPropInternal>,
]
`;
